lib/rules/no-typos.js

```diff
@@ -124,15 +124,18 @@ module.exports = {
       }
     }
 
-    function reportErrorIfClassPropertyCasingTypo(node, propertyName) {
+    function reportErrorIfPropertyCasingTypo(node, propertyName, isClassProperty) {
       if (propertyName === 'propTypes' || propertyName === 'contextTypes' || propertyName === 'childContextTypes') {
         checkValidPropObject(node);
       }
       STATIC_CLASS_PROPERTIES.forEach(CLASS_PROP => {
         if (propertyName && CLASS_PROP.toLowerCase() === propertyName.toLowerCase() && CLASS_PROP !== propertyName) {
+          const message = isClassProperty
+            ? 'Typo in static class property declaration'
```

Comment:
i think it's fine to have different messages, but "static class property" is a conceptual term prior to the feature landing in the language, so i think that'd be reasonable.